pandas/tests/extension/test_arrow.py

```diff
@@ -3330,6 +3330,19 @@ def test_factorize_chunked_dictionary():
     tm.assert_index_equal(res_uniques, exp_uniques)
 
 
+def test_factorize_dictionary_with_na():
+    # Test that factorize properly handles NA values in dictionary arrays
```

Comment:
Can you start the test with a comment referencing the issue, e.g.

    # GH#60567